hack/jenkins/e2e.sh

```diff
@@ -49,29 +53,48 @@ echo "E2E_OPT: ${E2E_OPT}"                         # hack/e2e.go options
 echo "E2E_SET_CLUSTER_API_VERSION: ${E2E_SET_CLUSTER_API_VERSION:-<not set>}" # optional, for GKE, set CLUSTER_API_VERSION to git hash
 echo "--------------------------------------------------------------------------------"
 
-# GCE variables
-export INSTANCE_PREFIX=${E2E_CLUSTER_NAME}
-export KUBE_GCE_ZONE=${E2E_ZONE}
-export KUBE_GCE_NETWORK=${E2E_NETWORK}
 
-# GKE variables
-export CLUSTER_NAME=${E2E_CLUSTER_NAME}
-export ZONE=${E2E_ZONE}
-export KUBE_GKE_NETWORK=${E2E_NETWORK}
+if [[ "${KUBERNETES_PROVIDER}" == "aws" ]]; then
+    export KUBE_AWS_INSTANCE_PREFIX=${E2E_CLUSTER_NAME}
+    export KUBE_AWS_ZONE=${E2E_ZONE}
+else
+    # GCE variables
+    export INSTANCE_PREFIX=${E2E_CLUSTER_NAME}
+    export KUBE_GCE_ZONE=${E2E_ZONE}
+    export KUBE_GCE_NETWORK=${E2E_NETWORK}
+
+    # GKE variables
+    export CLUSTER_NAME=${E2E_CLUSTER_NAME}
+    export ZONE=${E2E_ZONE}
+    export KUBE_GKE_NETWORK=${E2E_NETWORK}
+fi
+
 
 export PATH=${PATH}:/usr/local/go/bin
-export HOME=${WORKSPACE} # Nothing should want Jenkins $HOME
 export KUBE_SKIP_CONFIRMATIONS=y
 
-# sudo gcloud components update -q
+if [[ -d _output/ ]]; then
```

Comment:
I know this sounds paranoid, but I'd rather have this controlled by environment as well. The behavior below is very explicitly saying "if I find *anything* in the directory, bail out", because we're running the Jenkins "clean workspace plugin", so I'm asserting that it's working. Yet if somehow e2e tests were to generate an _output dir and the plugin were to fail, this condition could be met.